javalib/src/main/scala/java/util/zip/ZipEntry.scala

```diff
@@ -0,0 +1,287 @@
+    name.hashCode()
+
+}
+
+object ZipEntry extends ZipConstants {
+  final val DEFLATED: Int = 8
+  final val STORED: Int   = 0
+
+  private def myReadFully(in: InputStream, b: Array[Byte]): Unit = {
+    var len = b.length
+    var off = 0
+
+    while (len > 0) {
+      val count = in.read(b, off, len)
+      if (count <= 0) {
+        throw new EOFException()
+      }
+      off += count
+      len -= count
+    }
+  }
+
+  private[zip] def readIntLE(raf: RandomAccessFile): Long = {
+    val b0 = raf.read()
+    val b1 = raf.read()
+    val b2 = raf.read()
+    val b3 = raf.read()
+
+    if (b3 < 0) {
+      throw new EOFException()
+    } else {
+      b0 | (b1 << 8) | (b2 << 16) | (b3 << 24) // ATTENTION: DOES SIGN EXTENSION: IS THIS WANTED?
+    }
+  }
+
+  def fromInputStream(ler: LittleEndianReader, in: InputStream): ZipEntry = {
+    val hdrBuf = ler.hdrBuf
+    myReadFully(in, hdrBuf)
+
+    val sig = ((hdrBuf(0) & 0xFF) | ((hdrBuf(1) & 0xFF) << 8) | ((hdrBuf(2) & 0xFF) << 16) | ((hdrBuf(
+      3) & 0xFF) << 24)).toLong & 0xFFFFFFFFL
+    if (sig != CENSIG) {
+      throw new ZipException("Central Directory Entry not found")
+    }
+
+    val compressionMethod = (hdrBuf(10) & 0xff) | ((hdrBuf(11) & 0xff) << 8)
+    val time              = (hdrBuf(12) & 0xff) | ((hdrBuf(13) & 0xff) << 8)
+    val modDate           = (hdrBuf(14) & 0xff) | ((hdrBuf(15) & 0xff) << 8)
+    val crc = (hdrBuf(16) & 0xff) | ((hdrBuf(17) & 0xff) << 8) | ((hdrBuf(18) & 0xff) << 16) | ((hdrBuf(
+      19) << 24) & 0xffffffffL)
+    val compressedSize = (hdrBuf(20) & 0xff) | ((hdrBuf(21) & 0xff) << 8) | ((hdrBuf(
+      22) & 0xff) << 16) | ((hdrBuf(23) << 24) & 0xffffffffL)
+    val size = (hdrBuf(24) & 0xff) | ((hdrBuf(25) & 0xff) << 8) | ((hdrBuf(26) & 0xff) << 16) | ((hdrBuf(
+      27) << 24) & 0xffffffffL)
+    val nameLen    = (hdrBuf(28) & 0xff) | ((hdrBuf(29) & 0xff) << 8)
+    val extraLen   = (hdrBuf(30) & 0xff) | ((hdrBuf(31) & 0xff) << 8)
+    val commentLen = (hdrBuf(32) & 0xff) | ((hdrBuf(33) & 0xff) << 8)
+    val mLocalHeaderRelOffset = (hdrBuf(42) & 0xff) | ((hdrBuf(43) & 0xff) << 8) | ((hdrBuf(
+      44) & 0xff) << 16) | ((hdrBuf(45) << 24) & 0xffffffffL)
+
+    val nameBytes = new Array[Byte](nameLen)
+    myReadFully(in, nameBytes)
+
+    val commentBytes =
+      if (commentLen > 0) {
+        val commentBytes = new Array[Byte](commentLen)
+        myReadFully(in, commentBytes)
+        commentBytes
+      } else {
+        null
+      }
+
+    val extra =
+      if (extraLen > 0) {
+        val extra = new Array[Byte](extraLen)
+        myReadFully(in, extra)
+        extra
+      } else {
+        null
+      }
+
+    try {
+      /*
+       * The actual character set is "IBM Code Page 437".  As of
+       * Sep 2006, the Zip spec (APPNOTE.TXT) supports UTF-8.  When
+       * bit 11 of the GP flags field is set, the file name and
+       * comment fields are UTF-8.
+       *
+       * TODO: add correct UTF-8 support.
```

Comment:
Sure.
Just for the records, this comment comes from Harmony's source code:
https://github.com/apache/harmony/blob/java6/classlib/modules/archive/src/main/java/java/util/zip/ZipEntry.java#L404-L411